### PR TITLE
Fix direct check when mapping bug

### PR DESCRIPTION
**Pull request by tzheleznyak**
Bugfix to make test ONYX-10889 work.

Currently when the mapping is given we can check a host annotation directly and according to mapping. The bugfix insures error being thrown if one of the host annotations is a mapped claim.

### CODE REVIEW

File: app/domain/errors.rb [hunk 1]
```diff
@@ -584,7 +584,7 @@ module Constraints
       )
 
       NonPermittedRestrictionGiven = ::Util::TrackableErrorClass.new(
-        msg: "Role can't have one of these none permitted restrictions '{0-restrictions}'",
+        msg: "Role can't have standard claim or a mapped claim",
```
Comment (sashaCher) [inline app/domain/errors.rb:587]: This message a bit problematic for me.
It's thrown by infrastructural class [NonPermittedConstraint](https://github.com/cyberark/conjur/blob/0084b9d7f2a6b09eaa75093dd994f38ccee3e52d/app/domain/authentication/constraints/non_permitted_constraint.rb#L17) but contains the name of concrete features in it.
What will happen if any other authenticaficator will need to use the class?
What will happen if other parameters except mapped claims will affect permitted restrictions list?

Comment (sashaCher) [reply app/domain/errors.rb:587]: I prefer to keep the message as is as long it's an infrastructural message.
If we want to make things clear we should catch an exception and to have jet specific treatment for it.

File: app/domain/authentication/authn_jwt/restriction_validation/create_constraints.rb [hunk 2]
```diff
@@ -79,8 +77,14 @@ def add_required_constraint
           @constraints.append(required_constraint)
         end
 
+        def non_permitted_constraint
+          @non_permitted_constraint ||= @non_permitted_constraint_class.new(
+            non_permitted: @base_non_permitted_annotations + mapping_claims.keys
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/restriction_validation/create_constraints.rb:82]:
```suggestion
            non_permitted: @base_non_permitted_annotations + mapping_claims.values
```
Keys are annotations and values are claim names: `brach:ref`.

File: app/domain/errors.rb (hunk 1 above)
Comment (hilagross) [reply app/domain/errors.rb:587]: I think we should try to make it more understandable, the way it was it wasn't so clear

File: app/domain/authentication/authn_jwt/restriction_validation/create_constraints.rb (hunk 2 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/restriction_validation/create_constraints.rb:82]: But i invert it

Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/restriction_validation/create_constraints.rb:82]: image

File: app/domain/errors.rb [hunk 3]
```diff
@@ -542,6 +542,11 @@ module AuthnJwt
         msg: "Failed to parse mapping claims, {0-purpose} value '{1-claim-value}' appears twice or more",
         code: "CONJ00113E"
       )
+
+      RoleWithStandardOrMappedClaimError = ::Util::TrackableErrorClass.new(
+        msg: "Role can't have standard claim or a mapped claim. Error : '{0-error}'",
```
Comment (sashaCher) [inline app/domain/errors.rb:547]:
```suggestion
        msg: "Role can't have registered claim or a mapped claim. Error : '{0-error}'",
```
https://datatracker.ietf.org/doc/html/rfc7519#section-4.1

**Review by sashaCher — COMMENTED** [review]
Looks good.
Left one comment

**Review by sashaCher — APPROVED** [review]
LGTM
Message will be changed in docs PR

**Pull request merged**